src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx

```diff
@@ -43,13 +44,16 @@ export function SubmodelsOverviewCard(props: SubmodelsOverviewCardProps) {
         const id = reference.keys[0].value;
 
         try {
-            let fetchedSubmodelData: Submodel;
-            try {
-                fetchedSubmodelData = await getSubmodelById(id);
-            } catch (e) {
-                fetchedSubmodelData = await performSearchSubmodelFromAllRepos(id);
+            const submodelResponse = await getSubmodelById(id);
```

Comment:
IMO the try-catch block doesnt make sense anymore here, because the API functions dont throw errors anymore. IMO the entire catch block is unreachable right now.

Reply:
Well it certainly is not pretty but rethrowing should work here. The component needs some love but this is out of scope for this already large PR.

Reply:
created a ticket for this MNES-1294